File: lib_local/gallery/src/main/java/com/yanzhenjie/durban/model/AspectRatio.java
package com.yanzhenjie.durban.model;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.Nullable;

/**
 * 裁剪比例实体类
 * 作用：存储裁剪框的宽高比例、比例标题（如 1:1、4:3、16:9）
 * 实现 Parcelable 用于跨组件、页面传递数据
 */
public class AspectRatio implements Parcelable {
    // 比例标题（如"1:1"）
    @Nullable
    private final String mAspectRatioTitle;
    // 宽/高比例
    private final float mAspectRatioX;
    private final float mAspectRatioY;

    public AspectRatio(@Nullable String aspectRatioTitle, float aspectRatioX, float aspectRatioY) {
        mAspectRatioTitle = aspectRatioTitle;
        mAspectRatioX = aspectRatioX;
        mAspectRatioY = aspectRatioY;
    }

    @Nullable
    public String getAspectRatioTitle() {
        return mAspectRatioTitle;
    }

    public float getAspectRatioX() {
        return mAspectRatioX;
    }

    public float getAspectRatioY() {
        return mAspectRatioY;
    }

    protected AspectRatio(Parcel in) {
        mAspectRatioTitle = in.readString();
        mAspectRatioX = in.readFloat();
        mAspectRatioY = in.readFloat();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(mAspectRatioTitle);
        dest.writeFloat(mAspectRatioX);
        dest.writeFloat(mAspectRatioY);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<AspectRatio> CREATOR = new Creator<>() {
        @Override
        public AspectRatio createFromParcel(Parcel in) {
            return new AspectRatio(in);
        }

        @Override
        public AspectRatio[] newArray(int size) {
            return new AspectRatio[size];
        }
    };

}

File: lib_local/gallery/src/main/java/com/yanzhenjie/durban/model/AspectRatio.kt
package com.yanzhenjie.durban.model

import android.os.Parcel
import android.os.Parcelable

class AspectRatio : Parcelable {
    var mAspectRatioTitle: String? = null
        private set
    var mAspectRatioX: Float? = null
        private set
    var mAspectRatioY: Float? = null
        private set

    constructor(aspectRatioTitle: String?, aspectRatioX: Float?, aspectRatioY: Float?) {
        mAspectRatioTitle = aspectRatioTitle
        mAspectRatioX = aspectRatioX
        mAspectRatioY = aspectRatioY
    }

    constructor(parcel: Parcel) {
        mAspectRatioTitle = parcel.readString()
        mAspectRatioX = parcel.readValue(Float::class.java.classLoader) as? Float
        mAspectRatioY = parcel.readValue(Float::class.java.classLoader) as? Float
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(mAspectRatioTitle)
        parcel.writeValue(mAspectRatioX)
        parcel.writeValue(mAspectRatioY)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<AspectRatio> {
        override fun createFromParcel(parcel: Parcel): AspectRatio {
            return AspectRatio(parcel)
        }

        override fun newArray(size: Int): Array<AspectRatio?> {
            return arrayOfNulls(size)
        }
    }

}